librouteros/connections.py

```
@@ -22,7 +22,13 @@ def read(self, length):
         """
         data = bytearray()
         while len(data) != length:
-            data += self.sock.recv((length - len(data)))
+            tmp = None
+            try:
+                tmp = self.sock.recv((length - len(data)))
+            except:
```

This will catch all exceptions.